src/Components/ADT3DSceneBuilder/Internal/Behaviors/Widgets/WidgetBuilders/DataHistoryWidgetBuilder/Internal/TimeSeriesFormCallout.tsx

```diff
@@ -0,0 +1,173 @@
+    TextField,
+    useTheme
+} from '@fluentui/react';
+import { IDataHistoryBasicTimeSeries } from '../../../../../../../../Models/Types/Generated/3DScenesConfiguration-v1.0.0';
+import produce from 'immer';
+import {
+    getCardboardListCalloutComponentStyles,
+    getCardboardListCalloutStyles
+} from '../../../../../../../CardboardListCallout/CardboardListCallout.styles';
+import { useBehaviorFormContext } from '../../../../../../../../Models/Context/BehaviorFormContext/BehaviorFormContext';
+import ModelledPropertyBuilder from '../../../../../../../ModelledPropertyBuilder/ModelledPropertyBuilder';
+import { SceneBuilderContext } from '../../../../../../ADT3DSceneBuilder';
+import {
+    ModelledPropertyBuilderMode,
+    numericPropertyValueTypes,
+    PropertyExpression
+} from '../../../../../../../ModelledPropertyBuilder/ModelledPropertyBuilder.types';
+
+interface IProp {
+    calloutTarget: string;
+    series: IDataHistoryBasicTimeSeries;
+    onDismiss: () => void;
+    onPrimaryActionClick: (item: IDataHistoryBasicTimeSeries) => void;
+}
+
+enum Mode {
+    'Add',
+    'Edit'
+}
+
+const defaultSeries: IDataHistoryBasicTimeSeries = {
+    expression: '',
+    unit: ''
+};
+
+/** This callout component consists form for time series authoring to show in line chart */
+const TimeSeriesFormCallout: React.FC<IProp> = ({
+    calloutTarget,
+    series,
+    onDismiss,
+    onPrimaryActionClick
+}: IProp) => {
+    const [mode] = useState<Mode>(series ? Mode.Edit : Mode.Add);
+    const [
+        seriesToEdit,
+        setSeriesToEdit
+    ] = useState<IDataHistoryBasicTimeSeries>(defaultSeries);
+    const { t } = useTranslation();
+    const {
+        config,
+        sceneId,
+        adapter,
+        state: { selectedElements }
+    } = useContext(SceneBuilderContext);
+    const { behaviorFormState } = useBehaviorFormContext();
+
+    useEffect(() => {
+        setSeriesToEdit(series ? series : defaultSeries);
+    }, [series]);
+
+    const title =
+        mode === Mode.Add
+            ? t('widgets.dataHistory.form.timeSeries.add')
+            : t('widgets.dataHistory.form.timeSeries.edit');
+
+    const primaryActionLabel = series ? t('edit') : t('add');
+
+    const handlePropertyChange = useCallback(
+        (newPropertyExpression: PropertyExpression) => {
+            setSeriesToEdit(
+                produce((draft) => {
+                    draft.expression = newPropertyExpression.expression;
+                })
+            );
+        },
+        []
+    );
+
+    const handleUnitChange = useCallback((_event, unit: string) => {
+        setSeriesToEdit(
+            produce((draft) => {
+                draft.unit = unit;
+            })
+        );
+    }, []);
+
+    const handleLabelChange = useCallback((_event, label: string) => {
+        setSeriesToEdit(
+            produce((draft) => {
+                draft.label = label;
+            })
+        );
+    }, []);
+
+    const isFormValid = useMemo(() => {
```

Comment:
wondering if we should actually enforce that unit is always present, or just make unit and label both optional.  my preference is that unit and label are both optional personally

Reply:
yeah when thinking about more, unit may not be enforced. I will make it optional